compiler/src/dotty/tools/repl/ReplDriver.scala

```diff
@@ -272,7 +272,7 @@ class ReplDriver(settings: Array[String],
       (
         typeAliases.map("// defined alias " + _.symbol.showUser) ++
         defs.map(rendering.renderMethod) ++
-        vals.map(rendering.renderVal).flatten
+        vals.map(rendering.renderVal).distinct
```

Comment:
How is `scala> print("Hello"); print("Hello")` rendered with this fix?

Reply:
```
scala>  print("Hello"); print("Hello") 
HelloHello
scala>  
```

Reply:
What about `val foo = 2; (); val bar = 3`?

Reply:
```
scala> val foo = 2; (); val bar = 3 
val foo: Int = 2

val bar: Int = 3
scala>  
```
The sequence keeps the order. Maybe Units could be dropped if there's something else in the renderVal sequence.

Reply:
Yes, something like:
```scala
val renderedVals =
  vals.flatMap(rendering.renderVal)

val toRender =
  typeAliases.map("// defined alias " + _.symbol.showUser) ++
  defs.map(rendering.renderMethod) ++
  renderedVals

toRender.foreach(str => out.println(SyntaxHighlighting(str)))

if (renderedVals.isEmpty && vals.nonEmpty)
  out.println() // prints new line for side-effectful unit value (see #1369)
```
I would also be fine with a solution, that prints an additional new line for each command interpreted.

Reply:
Yeah, this unit based solution has many edge cases.

So something like this at ReplDriver.scala:195?
```scala
      case parsed: Parsed if parsed.trees.nonEmpty =>
        val newState = compile(parsed)
          .withHistory(parsed.sourceCode :: state.history)
          .newRun(compiler, rootCtx)
        out.println()
        newState
```